### PR TITLE
provide descriptive errors via JSON on invalid requests

### PR DESCRIPTION
**Pull request by soft-circles**

### CODE REVIEW

File: src/main/kotlin/GameController.kt [hunk 1]
```diff
@@ -11,16 +11,27 @@ class GameController : IController {
...
     override fun generateResponse(request: HttpRequest): HttpResponse {
         val httpResponse = HttpResponse()
-        val gameData = jsonParse.generateGameData(request.body.toString(Charset.defaultCharset()))
         try {
-            val board = tttEngineParser.generateBoard(gameData.board, gameData.input)
-            httpResponse.status = Status.OK
-            httpResponse.setBody(generateBody(board))
-        } catch (e: Exception) {
+            val gameData = jsonParse.generateGameData(request.body.toString(Charset.defaultCharset()))
+            try {
```
Comment (td-extreme) [inline src/main/kotlin/GameController.kt:16]: What happens if the move the user sends to play is already taken?

Comment (soft-circles) [reply src/main/kotlin/GameController.kt:16]: @td-extreme The API will return the same board it received.

Comment (td-extreme) [reply src/main/kotlin/GameController.kt:16]: Does it send back an error message as well, telling the user the move was invalid??

Comment (soft-circles) [reply src/main/kotlin/GameController.kt:16]: Hmmm, it doesn't send back an error message but I can definitely include that as part of the JSON. I still think it should send back the same board provided all the other input was correct

Comment (td-extreme) [reply src/main/kotlin/GameController.kt:16]: yeah, I would want to know why I got the same board back.

Comment (soft-circles) [reply src/main/kotlin/GameController.kt:16]: makes sense. I'll make that change

**Pull request merged**